R/process_lookup_homelessness.R

```diff
@@ -0,0 +1,114 @@
+# It would probably be simplest to create a lookup from the Homelessness data of one row per CHI +
+#   start dates of all applications. This can then be matched to episode and individual to easily create the hl1_in_fy flag(s)
+# (if someone is in the lookup -> 1, otherwise -> 0). For the episode file, it is slightly more complicated to create the other flags,
+# either matching on all the data by CHI and then doing some comparison of dates (as in SPSS),
+# or using some of the fancy new joins (dplyr 1.1.0 joins) and doing it that way.
+#
+#
+
+
+# data <- slfhelper::read_slf_episode("1718", c("anon_chi", "record_keydate1", "record_keydate2", "recid"))
+
+# year <- "1718"
+
+
+#' create homelessness lookup
+#' @description reads in homelessness extract and selects CHIs and flags them as homeless in FY
+#'
+#' @param year The year to process, in FY format.
+#' @return the final data as a [tibble][tibble::tibble-package].
+#' @export
+#' @family process extracts
+#'
+create_homelessness_lookup <- function(year) {
+  homelessness_lookup <- read_file(get_source_extract_path(year, "Homelessness")) %>%
+    dplyr::distinct(chi, record_keydate1, record_keydate2) %>%
+    tidyr::drop_na(chi) %>%
+    dplyr::mutate(hl1_in_fy = 1L)
+
+  return(homelessness_lookup)
+}
+
+
+#' add homelessness flag episode
+#' @description add homelessness in FY flag to episode/individual file
+#'
+#' @param data The extract to process
+#' @param year The year to process, in FY format.
+#'
+#' @return the final data as a [tibble][tibble::tibble-package].
+#' @export
+#'
+
+add_homelessness_flag <- function(data, year) {
+  lookup <- create_homelessness_lookup(year) %>%
+    slfhelper::get_anon_chi()
```

Comment:
1. `data` will have `chi` not `anon_chi`
2. We can have the lookup as a parameter, which will allow us to pass it through in the targets pipeline. This will create a 'dependency' (as with my other comment) but will also allow us to only create the lookup once and use it multiple times.

```suggestion
add_homelessness_flag <- function(data, year, lookup = create_homelessness_lookup(year)) { 
```